src/qibojit/tests/conftest.py

```diff
@@ -8,6 +8,10 @@
     # CI does not test for GPU
     _BACKENDS.append("cupy")
 
+if K._cuquantum_engine is not None:  # pragma: no cover
+    # CI does not test for GPU
+    _BACKENDS.append("cuquantum")
```

Comment:
This mechanism will not work as expected and we will not tests all backends, only numba and the default GPU backend. This is because once tests are launched in a machine with GPU, it will create the default GPU engine (currently cuquantum) and when it comes here the condition `if K._cupy_engine is not None` will not be satisfied and cupy will not be tested.

A potential fix would be:
```Python
_BACKENDS = ["numba"]
if K.gpu_devices:  # pragma: no cover
    # CI does not test for GPU
    _BACKENDS.extend(("cupy", "cuquantum"))
```
This way if a GPU is available we will test all backends.

Reply:
Yes, thank you.
I tested locally and the tests were performed using both cupy and cuquantum.
I've just push this change here 79926a5cb6c81f26f76723f9b2af35d45e57c6a1.